File: BOJ/python3/1764.py
'''
듣보잡
https://www.acmicpc.net/problem/1764
'''
import sys
import time
start = time.time()  # 시작 시간
sys.stdin = open("input.txt", "r")  # input.txt 파일로 입력

first = set()  # 명단 하나에만 있는 이름
second = []  # 두 명단 모두에 있는 이름

N, M = map(int, input().split())
for _ in range(N+M):
    name = input()

    if name not in first:
        first.add(name)
    else:
        second.append(name)

second.sort()
print(len(second))
print(*second, sep="\n")

# 현재시각 - 시작시간 = 실행 시간(sec)
print("--------------------")
print(f"Elapsed Time: {round(time.time() - start, 3)}s")

File: BOJ/nodejs/1764.js
/*
듣보잡
https://boj.kr/1764
 */
const start = new Date().getTime();

const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .split("\n");

const [N, M] = input[0].split(" ").map(Number); // [듣도 못한 사람수, 보도 못한 사람수]

const names = new Map();
for (let n = 0; n < N; n++) {
  names.set(input[1 + n]);
}
let cnt = 0;
const res = new Array();
for (let m = 0; m < M; m++) {
  if (names.has(input[1 + N + m])) {
    cnt++;
    res.push(input[1 + N + m]);
  }
}
const ans = cnt + "\n" + res.sort().join("\n");

console.log(ans);

const end = new Date().getTime();
console.log("--------------------");
console.log(`Elapsed Time: ${(new Date().getTime() - start) / 1000}s`);
